plugins/woocommerce/includes/data-stores/class-wc-product-variable-data-store-cpt.php

```diff
@@ -710,4 +728,104 @@ public function untrash_variations( $product_id ) {
 
 		delete_transient( 'wc_product_children_' . $product_id );
 	}
+
+	/**
+	 * Validate the children data by checking the structure and type of the data.
+	 *
+	 * @param array  $children The children data.
+	 * @param string $current_version The current transient version.
+	 * @return bool True if valid, false otherwise.
+	 */
+	protected function validate_children_data( $children, $current_version ) {
+		if ( ! is_array( $children ) ) {
+			return false;
+		}
+
+		// Basic structure checks.
+		if ( ! isset( $children['all'] ) || ! isset( $children['visible'] ) ) {
+			return false;
+		}
+
+		// Version check - only if version is set.
+		if ( isset( $children['version'] ) && $children['version'] !== $current_version ) {
+			return false;
+		}
+
+		if ( ! is_array( $children['all'] ) || ! is_array( $children['visible'] ) ) {
+			return false;
+		}
+
+		foreach ( $children['all'] as $id ) {
+			if ( ! is_numeric( $id ) ) {
+				return false;
+			}
+		}
+
+		foreach ( $children['visible'] as $id ) {
+			if ( ! is_numeric( $id ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validate the prices data by checking the structure and type of the data.
+	 *
+	 * @param  array  $prices_array The prices data.
+	 * @param  string $current_version The current version of the data.
+	 * @return bool True if valid, false otherwise.
+	 */
+	protected function validate_prices_data( $prices_array, $current_version ) {
+		if ( ! is_array( $prices_array ) ) {
+			return false;
+		}
+
+		// Version check - only if version is set.
+		if ( isset( $prices_array['version'] ) && $prices_array['version'] !== $current_version ) {
+			return false;
+		}
+
+		// Remove version key for iteration.
+		$data_without_version = array_diff_key( $prices_array, array( 'version' => '' ) );
+
+		// Iterate through each price hash entry.
+		foreach ( $data_without_version as $price_data ) {
+			if ( ! is_array( $price_data ) ) {
+				return false;
+			}
+
+			$required_types = array( 'price', 'regular_price', 'sale_price' );
+			foreach ( $required_types as $type ) {
+				if ( ! isset( $price_data[ $type ] ) || ! is_array( $price_data[ $type ] ) ) {
+					return false;
+				}
+			}
+
+			// Validate individual prices.
+			foreach ( $price_data['price'] as $variation_id => $price ) {
+				if ( ! is_numeric( $variation_id ) ) {
+					return false;
+				}
+
+				if ( ! is_numeric( $price ) && '' !== $price ) {
+					return false;
+				}
+
+				foreach ( $required_types as $type ) {
+					if ( ! array_key_exists( $variation_id, $price_data[ $type ] ) ) {
+						return false;
+					}
+
+					$type_price = $price_data[ $type ][ $variation_id ];
+					if ( ! is_numeric( $type_price ) && '' !== $type_price ) {
+						return false;
+					}
+				}
```

Comment:
I might be misunderstanding this, but I think we are checking `price` twice. One inside the `if` (L812-814) and another one inside the `foreach` (L822-824). Could one of them be removed?

Reply:
Nope you were right, goot catch. Done in https://github.com/woocommerce/woocommerce/pull/53603/commits/c6c1801c0f9548b3367bd05e410c3254e01055fa